### PR TITLE
Add BindingInCode C++/CX Example & Project

### PR DESCRIPTION
**Pull request by michael-hawker**
Noticed there was no C++/CX example for binding in code behind, which is not trivial to convert from a C# example.

Note: Would be nice of root project README.md had instructions for adding samples and checking references (i.e. generating a preview of the MSDN doc from the md file).  It seems like there's extra infrastructure and resources that are missing when trying to load/run some projects.

### CODE REVIEW

File: windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+MainPage::MainPage()
+{
+	InitializeComponent();
+
+	//<SnippetBindingObject>
+	// Create the source string.
+	String^ s = ref new String(L"Hello");
+
+	// Create the binding description.
+	Binding^ b = ref new Binding();
```
Comment (ChrisGuzak) [inline windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp:33]: use auto here
auto b = ref new Binding()

File: windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+MainPage::MainPage()
+{
+	InitializeComponent();
+
+	//<SnippetBindingObject>
+	// Create the source string.
+	String^ s = ref new String(L"Hello");
```
Comment (ChrisGuzak) [inline windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp:30]: auto here
auto s = ref new String...

Comment (ChrisGuzak) [reply windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp:30]: and I'd inline this below, no need for the temporary "s"

File: windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp (hunk 1 above)
Comment (ChrisGuzak) [reply windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp:33]: inline this temporary
MyText->SetBinding(MyTest->TextProperty, ref new Binding());

**Comment by michael-hawker** [comment]
For all comments, I'm mimicking the C# example [here](https://github.com/MicrosoftDocs/winrt-api/blob/docs/windows.ui.xaml/code/BindingInCode/csharp/Page.xaml.cs).

This is a basic example, so I'd think we don't want to over optimize with auto?  I don't think I've seen that elsewhere.  It's important for a developer to see the explicit type with the ^ notation too and the parity with the C# example for comparison.

File: windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp (hunk 1 above)
Comment (michael-hawker) [reply windows.ui.xaml/code/BindingInCode/cpp/MainPage.xaml.cpp:33]: The binding needs properties set, so it can't be inlined.

**Review by jwmsft — APPROVED** [review]

**Pull request merged**